ide/textmate.lexer/arch.xml

```diff
@@ -0,0 +1,1115 @@
+            should be covered by the tests? How you find out that your
+            project was successful?
+            </hint>
+        </question>
+-->
+ <answer id="arch-quality">
+  <p>
+   Unit tests for the lexer.
+  </p>
+ </answer>
+
+
+
+<!--
+        <question id="arch-time" when="init">
+            What are the time estimates of the work?
+            <hint>
+            Please express your estimates of how long the design, implementation,
+            stabilization are likely to last. How many people will be needed to
+            implement this and what is the expected milestone by which the work should be 
+            ready?
+            </hint>
+        </question>
+-->
+ <answer id="arch-time">
+  <p>
+   Unclear.
+  </p>
+ </answer>
+
+
+
+<!--
+        <question id="arch-usecases" when="init">
+            <hint>
+                Content of this answer will be displayed as part of page at
+                http://www.netbeans.org/download/dev/javadoc/usecases.html 
+                You can use tags &lt;usecase name="name&gt; regular html description &lt;/usecase&gt;
+                and if you want to use an URL you can prefix if with @TOP@ to begin
+                at the root of your javadoc
+            </hint>
+        
+            Describe the main <a href="http://wiki.netbeans.org/API_Design#The_Importance_of_Being_Use_Case_Oriented">
+            use cases</a> of the new API. Who will use it under
+            what circumstances? What kind of code would typically need to be written
+            to use the module?
+        </question>
+-->
+ <answer id="arch-usecases">
+  <p>
+      <usecase id="register-grammar" name="Register grammar">
+          To register a new TextMate grammar, add it as a file into the system filesystem,
+          under "Editors/&lt;mime-type&gt;/", and make sure it has an attribute "textmate-grammar",&lt;/mime-type&gt;,
+          whose value is a string representing the TextMate top-level scope. For example:
+          <pre>
+    &lt;folder name="Editors"&gt;
+        &lt;folder name="text"&gt;
+            &lt;folder name="x-kotlin"&gt;
+                &lt;file name="Kotlin.tmLanguage.json" url="Kotlin.tmLanguage.json"&gt;
+                    &lt;attr name="textmate-grammar" stringvalue="source.kotlin" /&gt;
+                &lt;/file&gt;
+            &lt;/folder&gt;
+        &lt;/folder&gt;
+    &lt;/folder&gt;
+          </pre>
+      </usecase>
+  </p>
+ </answer>
+
+
+
+<!--
+        <question id="arch-what" when="init">
+            What is this project good for?
+            <hint>
+            Please provide here a few lines describing the project, 
+            what problem it should solve, provide links to documentation, 
+            specifications, etc.
+            </hint>
+        </question>
+-->
+ <answer id="arch-what">
+  <p>
+   This project to allows to easily add syntax highlighting for languages by
+   using the TextMate grammar files.
```

Comment:
Having a module that would provide some UI to actually add/download the grammar files might be also very nice.

Reply:
Great idea! Looking forward for the PR!